Sources/Networking/AlamofireNetworkClient/OAuth/OAuthInterceptor.swift

```diff
@@ -28,7 +28,7 @@ open class OAuthRequestInterceptor {
   private let provider: OAuthProvider
   private let storage: OAuthStorage
   private let adapter: OAuthHeadersAdapter
-  private let lock: NSLock = .init()
+  private let lock: DispatchSemaphore = .init(value: 1)
```

Comment:
Maybe we should rename the variable name as well and change it from `lock` to  `semaphore`.

Reply:
I think it’s okay because we are using semaphore as a lock.